libs9s/s9srpcclient.cpp

```diff
@@ -4454,6 +4459,91 @@ S9sRpcClient::registerPostgreSql(
     return executeRequest(uri, request);
 }
 
+/**
+ * \param hosts the hosts that will be the member of the cluster (variant list
+ *   with S9sNode elements).
+ * \param osUserName the user name to be used to SSH to the host.
+ * \returns true if the request sent and a return is received (even if the reply
+ *   is an error message).
+ *
+ * This method will create a job that creates a logical PostgreSQL cluster of
+ * streaming PostgreSQL clusters.
+ */
+bool
+S9sRpcClient::createLogicalPostgreSql(
+        const S9sVariantList &hosts,
+        const S9sString      &osUserName,
+        const S9sString      &psqlVersion)
+{
+    S9sOptions     *options = S9sOptions::instance();
+    S9sVariantMap   request;
+    S9sVariantMap   job = composeJob();
+    S9sVariantMap   jobData = composeJobData();
+    S9sVariantMap   jobSpec;
+    S9sString       uri = "/v2/jobs/";
+
+    if (hosts.size() < 1u && !options->hasClustersOption())
+    {
+        PRINT_ERROR("Either node list or clusters list should be specified when creating PostgreSQL logical cluster.");
+        return false;
+    }
+
+    addCredentialsToJobData(jobData);
+
+    //
+    // The job_data describing the cluster.
+    //
+    jobData["cluster_type"]     = "postgresql_logical";
+    jobData["type"]             = "postgresql";
+    jobData["vendor"]           = options->vendor();
+    if (options->hasEnterpriseToken())
+        jobData["enterprise_token"] = options->enterpriseToken();
+    jobData["nodes"]            = nodesField(hosts);
+    jobData["version"]          = psqlVersion;
+    jobData["postgre_user"]     = options->dbAdminUserName();
+    jobData["postgre_password"] = options->dbAdminPassword();
+    jobData["disable_firewall"] = !options->keepFirewall();
+    jobData["deploy_agents"]    = false;
+
+    if (options->withTimescaleDb())
+        jobData["install_timescaledb"] = true;
+
+    if (options->noInstall())
+    {
+        jobData["install_software"] = false;
+        jobData["enable_uninstall"] = false;
+    } else {
+        jobData["install_software"] = true;
+        jobData["enable_uninstall"] = false;
+    }
+
+    if (options->hasClustersOption())
```

Comment:
I think we should parse that string here and verify if it is valid and contains at least 2 CIDs.

Reply:
Yes, would be nice, I just thought it to be optional.

Reply:
Actually, as I am about to adjust this, may be better not too. The option name is just "clusters" now. Clusters not only have identifier number but name as well. In many use case the cluster name can be used as well.
So it is better not to parse this here in my opinion. On the cmon side we can look for cluster names as well if we get there to implement it there.